sgl-kernel/csrc/moe/moe_fused_gate.cu

```diff
@@ -219,12 +222,40 @@ __device__ void moe_fused_gate_impl(
     __syncthreads();
   }
 
+  // Handle n_share_experts_fusion if needed
+  if (n_share_experts_fusion > 0 && thread_group_idx == 0) {
+    // For the last position, we need to set a random expert ID and adjust the weight
+    int64_t last_idx = topk * thread_row + (topk - 1);
+
+    // Use curand to generate random number
+    curandStatePhilox4_32_10_t state;
+    curand_init(thread_row, 0, 0, &state);
+    int64_t random_offset = curand(&state) % n_share_experts_fusion;
+    indices_ptr[last_idx] = static_cast<int32_t>(params.NUM_EXPERTS + random_offset);
+
+    // Calculate the sum of the first k-1 weights
+    float prev_sum = 0.0f;
+    for (int ii = 0; ii < topk - 1; ++ii) {
+      prev_sum += output_ptr[topk * thread_row + ii];
+    }
+
+    __syncthreads();
+
+    // Set the weight to the sum of the first k-1 weights divided by 2.5 (as in the Python implementation)
+    output_ptr[last_idx] = prev_sum / 2.5f;
+
+    // For renormalization, we only use the sum of the first k-1 weights
```

Comment:
if I understand correctly, it seems currently we compute topk:=8+1=9 things, and then override `output_sum` as well as `data[-1]`. then, what about we only compute 8 things, then it is a bit cheaper (since we do not compute 9th top thing), and we do not need to recompute output_sum.

Reply:
done.